devtempstats_service/native/src/impl/DevTempStatsService.cpp

```diff
@@ -0,0 +1,272 @@
+#include "DevTempStatsService.hpp"
+
+#include <utils/Log.h>
+
+#include <vector>
+#include <fstream>
+#include <string>
+#include <sstream>
+#include <cstdlib>
+#include <iostream>
+
+#ifdef LOG_TAG
+#undef LOG_TAG
+#endif
+#define LOG_TAG "devtempstats_service"
+
+namespace devtempstats
+{
+   namespace service
+   {
+
+      ::std::shared_ptr<DevTempStatsService> DevTempStatsService::S_INSTANCE = NULL;
+
+      ::std::shared_ptr<DevTempStatsService> DevTempStatsService::getInstance()
```

Comment:
Aceasta varianta de singleton poate produce un race condition atunci cand 2 thread-uri executa functia `getInstance` in paralel.
Putem evita aceasta problema mutand initializarea lui `S_INSTANCE` in function `getInstance` ca "static function variable". Atunci, vom avea garantia ca variabla va fi initializata o singura data: https://stackoverflow.com/a/8102145

e.g.
```suggestion
      ::std::shared_ptr<DevTempStatsService> DevTempStatsService::getInstance() {
      static ::std::shared_ptr<DevTempStatsService> DevTempStatsService::S_INSTANCE = ndk::SharedRefBase::make<DevTempStatsService>();
      return S_INSTANCE;
      }
```